File: SocialMeet/src/main/java/com/example/socialmeet/entity/CallSession.java
package com.example.socialmeet.entity;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "call_sessions")
public class CallSession {
    
    @Id
    @Column(name = "id")
    private String id;
    
    @Column(name = "caller_id", nullable = false)
    private Long callerId;
    
    @Column(name = "receiver_id", nullable = false)
    private Long receiverId;
    
    @Column(name = "status", nullable = false)
    private String status; // INITIATED, RINGING, ACTIVE, ENDED, REJECTED, CANCELLED, FAILED
    
    @Column(name = "start_time")
    private LocalDateTime startTime;
    
    @Column(name = "end_time")
    private LocalDateTime endTime;
    
    @Column(name = "duration")
    private Long duration; // 通话时长（秒）
    
    @Column(name = "rate", precision = 10, scale = 2)
    private BigDecimal rate; // 每分钟费率（元）
    
    @Column(name = "total_cost", precision = 10, scale = 2)
    private BigDecimal totalCost; // 总费用（元）
    
    @Column(name = "caller_balance", precision = 10, scale = 2)
    private BigDecimal callerBalance; // 发起方余额
    
    @Column(name = "receiver_balance", precision = 10, scale = 2)
    private BigDecimal receiverBalance; // 接收方余额
    
    @Column(name = "is_online")
    private Boolean isOnline; // 接收方是否在线
    
    @Column(name = "created_at")
    private LocalDateTime createdAt;
    
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
    
    // 构造函数
    public CallSession() {
        this.createdAt = LocalDateTime.now();
        this.updatedAt = LocalDateTime.now();
    }
    
    // Getters and Setters
    public String getId() {
        return id;
    }
    
    public void setId(String id) {
        this.id = id;
    }
    
    public Long getCallerId() {
        return callerId;
    }
    
    public void setCallerId(Long callerId) {
        this.callerId = callerId;
    }
    
    public Long getReceiverId() {
        return receiverId;
    }
    
    public void setReceiverId(Long receiverId) {
        this.receiverId = receiverId;
    }
    
    public String getStatus() {
        return status;
    }
    
    public void setStatus(String status) {
        this.status = status;
    }
    
    public LocalDateTime getStartTime() {
        return startTime;
    }
    
    public void setStartTime(LocalDateTime startTime) {
        this.startTime = startTime;
    }
    
    public LocalDateTime getEndTime() {
        return endTime;
    }
    
    public void setEndTime(LocalDateTime endTime) {
        this.endTime = endTime;
    }
    
    public Long getDuration() {
        return duration;
    }
    
    public void setDuration(Long duration) {
        this.duration = duration;
    }
    
    public BigDecimal getRate() {
        return rate;
    }
    
    public void setRate(BigDecimal rate) {
        this.rate = rate;
    }
    
    public BigDecimal getTotalCost() {
        return totalCost;
    }
    
    public void setTotalCost(BigDecimal totalCost) {
        this.totalCost = totalCost;
    }
    
    public BigDecimal getCallerBalance() {
        return callerBalance;
    }
    
    public void setCallerBalance(BigDecimal callerBalance) {
        this.callerBalance = callerBalance;
    }
    
    public void setCallerBalance(Double callerBalance) {
        this.callerBalance = BigDecimal.valueOf(callerBalance);
    }
    
    public BigDecimal getReceiverBalance() {
        return receiverBalance;
    }
    
    public void setReceiverBalance(BigDecimal receiverBalance) {
        this.receiverBalance = receiverBalance;
    }
    
    public void setReceiverBalance(Double receiverBalance) {
        this.receiverBalance = BigDecimal.valueOf(receiverBalance);
    }
    
    public Boolean getIsOnline() {
        return isOnline;
    }
    
    public void setIsOnline(Boolean isOnline) {
        this.isOnline = isOnline;
    }
    
    public LocalDateTime getCreatedAt() {
        return createdAt;
    }
    
    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }
    
    public LocalDateTime getUpdatedAt() {
        return updatedAt;
    }
    
    public void setUpdatedAt(LocalDateTime updatedAt) {
        this.updatedAt = updatedAt;
    }
    
    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: app/src/main/java/com/example/myapplication/model/CallSession.kt
package com.example.myapplication.model

import com.google.gson.annotations.SerializedName
import java.util.Date

/**
 * 通话会话数据模型
 */
data class CallSession(
    @SerializedName("id")
    val id: String = "",
    
    @SerializedName("callerId")
    val callerId: Long = 0,
    
    @SerializedName("receiverId")
    val receiverId: Long = 0,
    
    @SerializedName("status")
    val status: CallStatus = CallStatus.INITIATED,
    
    @SerializedName("startTime")
    val startTime: Date? = null,
    
    @SerializedName("endTime")
    val endTime: Date? = null,
    
    @SerializedName("duration")
    val duration: Long = 0, // 通话时长（秒）
    
    @SerializedName("rate")
    val rate: Double = 300.0, // 每分钟费率（元）
    
    @SerializedName("totalCost")
    val totalCost: Double = 0.0, // 总费用（元）
    
    @SerializedName("callerBalance")
    val callerBalance: Double = 0.0, // 发起方余额
    
    @SerializedName("receiverBalance")
    val receiverBalance: Double = 0.0, // 接收方余额
    
    @SerializedName("isOnline")
    val isOnline: Boolean = false, // 接收方是否在线
    
    @SerializedName("createdAt")
    val createdAt: Date = Date(),
    
    @SerializedName("updatedAt")
    val updatedAt: Date = Date()
) {
    /**
     * 获取通话状态文本
     */
    fun getStatusText(): String {
        return when (status) {
            CallStatus.INITIATED -> "发起中"
            CallStatus.RINGING -> "响铃中"
            CallStatus.ACTIVE -> "通话中"
            CallStatus.ENDED -> "已结束"
            CallStatus.REJECTED -> "已拒绝"
            CallStatus.CANCELLED -> "已取消"
            CallStatus.FAILED -> "失败"
        }
    }
    
    /**
     * 获取通话状态颜色
     */
    fun getStatusColor(): String {
        return when (status) {
            CallStatus.INITIATED -> "#FF9800"
            CallStatus.RINGING -> "#2196F3"
            CallStatus.ACTIVE -> "#4CAF50"
            CallStatus.ENDED -> "#9E9E9E"
            CallStatus.REJECTED -> "#F44336"
            CallStatus.CANCELLED -> "#FF5722"
            CallStatus.FAILED -> "#F44336"
        }
    }
    
    /**
     * 获取通话时长文本
     */
    fun getDurationText(): String {
        val minutes = duration / 60
        val seconds = duration % 60
        return String.format("%02d:%02d", minutes, seconds)
    }
    
    /**
     * 获取费用文本
     */
    fun getCostText(): String {
        return String.format("¥%.2f", totalCost)
    }
    
    /**
     * 获取费率文本
     */
    fun getRateText(): String {
        return String.format("¥%.0f/分钟", rate)
    }
    
    /**
     * 检查是否可以发起通话
     */
    fun canInitiateCall(): Boolean {
        return isOnline && callerBalance >= rate / 60.0 // 至少能支付1秒的费用
    }
    
    /**
     * 检查是否可以继续通话
     */
    fun canContinueCall(): Boolean {
        return status == CallStatus.ACTIVE && callerBalance >= rate / 60.0
    }
}

/**
 * 通话状态枚举
 */
enum class CallStatus {
    INITIATED,    // 发起中
    RINGING,      // 响铃中
    ACTIVE,       // 通话中
    ENDED,        // 已结束
    REJECTED,     // 已拒绝
    CANCELLED,    // 已取消
    FAILED        // 失败
}
